src/database/migrations/0066-create-opportunity-permissions-tables.sql

```diff
@@ -0,0 +1,28 @@
+CREATE TYPE opportunity_permission_t AS ENUM (
+	'manage',
+	'edit',
+	'view',
+	'create_proposal'
+);
+
+CREATE TABLE user_opportunity_permissions (
+	user_keycloak_user_id uuid NOT NULL REFERENCES users (
+		keycloak_user_id
+	) ON DELETE CASCADE,
+	opportunity_permission opportunity_permission_t NOT NULL,
+	opportunity_id int NOT NULL REFERENCES opportunities (id) ON DELETE CASCADE,
+	created_by uuid NOT NULL REFERENCES users (keycloak_user_id) ON DELETE CASCADE,
+	created_at timestamp with time zone NOT NULL DEFAULT now(),
+	not_after timestamp with time zone DEFAULT NULL,
+	PRIMARY KEY (user_keycloak_user_id, opportunity_id, opportunity_permission)
+);
+
+CREATE TABLE user_group_opportunity_permissions (
```

Comment:
I just realized that we didn't run the audit log function on this.

I do think we should add a migration that sets up that function as a trigger on table create....

Reply:
OK fixed!